partition-manager/generation/tests/access.rs

```diff
@@ -0,0 +1,244 @@
+use std::collections::{BTreeMap, BTreeSet};
+
+use partition_manager_generation::{Access, Disk, GeneratedPartition, Manifest, Partition, Variant};
+
+fn create_manifest() -> Manifest {
+    Manifest {
+        variants: BTreeSet::from_iter(["bootloader".into(), "application".into()]),
+        disk: Disk {
+            size: None,
+            alignment: None,
+        },
+        partitions: [(
+            "l1".to_string(),
+            Partition {
+                access: BTreeMap::new(),
+                offset: 0x0000,
+                size: 0x0100,
+            },
+        )]
+        .into(),
+    }
+}
+
+fn create_generated_manifest() -> Vec<GeneratedPartition> {
+    vec![GeneratedPartition {
+        name: "l1".to_string(),
+        access: partition_manager_generation::Access::RW,
+        offset: 0x0000,
+        size: 0x0100,
+    }]
+}
+
+#[test]
+fn variant_none() {
+    let result = create_manifest().generate(None).map(Vec::from_iter).unwrap();
+    assert_eq!(result, create_generated_manifest());
+}
+
+#[test]
+fn variant_unspecified() {
+    let result = create_manifest()
+        .generate(Some("Unspecified".to_string()))
+        .map(Vec::from_iter);
+
+    assert_eq!(
+        format!("{:?}", result),
+        "Err(Variant 'Unspecified' not defined in manifest)"
+    );
+}
+
+#[test]
+fn variant_no_match() {
+    let mut manifest = create_manifest();
+    manifest.partitions.get_mut("l1").unwrap().access = BTreeMap::from_iter([("bootloader".into(), Access::RW)]);
+
+    let result = manifest
+        .generate(Some("application".to_string()))
+        .map(Vec::from_iter)
+        .unwrap();
+
+    assert_eq!(result, Vec::new());
```

Comment:
I really like the use of `assert_eq!` over `assert!(result.is_empty())` to get better output on error :+1: